crates/astria-core/src/protocol/abci.rs

```diff
@@ -1,76 +1,49 @@
 }
 
 impl AbciErrorCode {
+    /// Returns the wrapped [`Code`].
     #[must_use]
-    pub fn info(self) -> Cow<'static, str> {
-        match self.0 {
-            0 => "unspecified".into(),
-            1 => "provided path is unknown".into(),
-            2 => "one or more path parameters were invalid".into(),
-            3 => "an internal server error occurred".into(),
-            4 => "the provided nonce was invalid".into(),
-            5 => "the provided transaction was too large".into(),
-            6 => "insufficient funds".into(),
-            7 => "the provided chain id was invalid".into(),
-            8 => "the requested value was not found".into(),
-            9 => "the transaction expired in the app's mempool".into(),
-            10 => "the transaction failed to execute in prepare_proposal()".into(),
-            11 => "the request payload was malformed".into(),
-            other => format!("unknown non-zero abci error code: {other}").into(),
-        }
+    pub const fn value(self) -> Code {
+        self.0
     }
 }
 
 impl std::fmt::Display for AbciErrorCode {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{}", self.info())
-    }
-}
-
-impl From<AbciErrorCode> for tendermint::abci::Code {
-    fn from(value: AbciErrorCode) -> Self {
-        value.0.into()
-    }
-}
-
-impl From<NonZeroU32> for AbciErrorCode {
-    fn from(value: NonZeroU32) -> Self {
-        match value.get() {
-            1 => Self::UNKNOWN_PATH,
-            2 => Self::INVALID_PARAMETER,
-            3 => Self::INTERNAL_ERROR,
-            4 => Self::INVALID_NONCE,
-            5 => Self::TRANSACTION_TOO_LARGE,
-            6 => Self::INSUFFICIENT_FUNDS,
-            7 => Self::INVALID_CHAIN_ID,
-            8 => Self::VALUE_NOT_FOUND,
-            9 => Self::TRANSACTION_EXPIRED,
-            10 => Self::TRANSACTION_FAILED,
-            11 => Self::BAD_REQUEST,
-            other => Self(other),
-        }
+        f.write_str(match *self {
+            Self::UNKNOWN_PATH => "provided path is unknown",
+            Self::INVALID_PARAMETER => "one or more path parameters were invalid",
+            Self::INTERNAL_ERROR => "an internal server error occurred",
+            Self::INVALID_NONCE => "the provided nonce was invalid",
+            Self::TRANSACTION_TOO_LARGE => "the provided transaction was too large",
+            Self::INSUFFICIENT_FUNDS => "insufficient funds",
+            Self::INVALID_CHAIN_ID => "the provided chain id was invalid",
+            Self::VALUE_NOT_FOUND => "the requested value was not found",
+            Self::TRANSACTION_EXPIRED => "the transaction expired in the app's mempool",
+            Self::TRANSACTION_FAILED => "the transaction failed to execute in prepare_proposal()",
+            Self::BAD_REQUEST => "the request payload was malformed",
+            _ => "invalid error code: should be unreachable (this is a bug)",
```

Comment:
This should report the actual number

Reply:
Happy to add the number, but there's currently no possible way to actually construct an invalid `AbciErrorCode`.  I guess we could add a `Deserialize` constructor in the future which supports constructing an invalid one or something though, so makes sense to add the value here.

Reply:
Done in [5367a02](https://github.com/astriaorg/astria/pull/1280/commits/5367a0230be034ed20550471579909ba6746b604).